### ISSUE TITLE
Crash with different warrior types

### ISSUE THREAD

**Issue by dougajmcdonald** [gareththegeek/corewar#86]
I loaded a dwarf and gemini, init'd and hit run and all I got with this lousy error

![image](https://user-images.githubusercontent.com/1190042/33683766-f17f4daa-dac3-11e7-97d0-fb34560910fe.png)

**Comment by dougajmcdonald** [comment]
It's the gemini warrior, I tried two of them and got the same error:

```
;redcode
;name Gemini
;author A. K. Dewdney
;assert 1

	DAT		0
	DAT		99
start	MOV	@-2,	@-1
	CMP	-3,	#9
	JMP	4
	ADD	#1,	-5
	ADD	#1,	-5
	JMP	-5
	MOV	#99,	93
	JMP	93

	END	start
```

**Comment by dougajmcdonald** [comment]
I get the same error but later on when I run two mice:

```
;redcode
;name MICE
;author Chip Wendell
;
;       MICE
;       by:
;         Chip Wendell
;
;
;        org     org
ptr     dat     #0     ,  #0
start   mov     #12    ,  ptr
loop    mov     @ptr   ,  <copy
        djn     loop   ,  ptr
        spl     @copy  ,  0
        add     #653   ,  copy
        jmz     start  ,  ptr
copy    dat     #0     ,  #833
        end     start
```

**Comment by gareththegeek** [comment]
Resolved by #91
